### PR TITLE
lend-35-polygon-support

### PR DESCRIPTION
**Pull request by tracyarciaga**
## Description of Changes

Added halo-matic markets and configs and deploys scripts for Polygon 

[Link to Jira Ticket](https://halodao.atlassian.net/browse/LEND-35)

### Developer Checklist:

- [x] I have followed the guidelines in our Contributing document
- [x] This PR has a corresponding JIRA ticket
- [x] My branch conforms with our naming convention i.e. `feature/HDEV-XXX-description`
- [x] I have written new tests for your core changes, as applicable
- [x] I have successfully ran tests locally
- [x] I have formatted my code using format document in VSCode

### Reviewers Checklist:

- [ ] Code is readable and understandable; any unclear parts have explanations
- [ ] UI/UX changes match the corresponding figma/other design resources, if applicable
- [ ] I have successfully ran tests locally

### CODE REVIEW

**Review by 0xAplki — COMMENTED** [review]
I guess to fix your error we can add atleast 1 GWEI on maxPriorityFee

Here's my proposed workaround:
1 - add a const for `NETWORK_DEFAULT_PRIORITYFEE` and pattern in like `NETWORKS_DEFAULT_GAS`. set everything to 1 * GWEI (found in `helper-hardhat-config.ts`)
2 - add  `maxPriorityFeePerGas` param in `getCommonNetworkConfig` on `hardhat.config.ts` with the value `NETWORK_DEFAULT_PRIORITYFEE[networkName]`


Let see if this works!

File: markets/halo-matic/commons.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  },
+  WrappedNativeToken: {
+    [ePolygonNetwork.mumbai]: '0x9c3C9283D3e44854697Cd22D3Faa240Cfb032889',
+    [ePolygonNetwork.matic]: '0x0d500B1d8E8eF31E21C99d1Db9A6444d3ADf1270',
+  },
+  ReserveFactorTreasuryAddress: {
+    [ePolygonNetwork.mumbai]: ZERO_ADDRESS,
+    [ePolygonNetwork.matic]: '0x7734280A4337F37Fbf4651073Db7c28C80B339e9',
+  },
+  IncentivesController: {
```
Comment (0xAplki) [inline markets/halo-matic/commons.ts:158]: can we have this zero for now since we dont have it

File: markets/halo-matic/constants.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { ZERO_ADDRESS } from '../../helpers/constants';
+import { ePolygonNetwork } from '../../helpers/types';
+
+export const HALO_CONTRACT_ADDRESSES = {
+  [ePolygonNetwork.matic]: {
+    rewardToken: '0x47BE779De87de6580d0548cde80710a93c502405',
+    emissionManager: '0x235A2ac113014F9dcb8aBA6577F20290832dDEFd',
+    lendingPoolAddress: '0xC73b2c6ab14F25e1EAd3DE75b4F6879DEde3968E',
```
Comment (0xAplki) [inline markets/halo-matic/constants.ts:8]: we already deployed this?

**Review by 0xAplki — APPROVED** [review]

**Comment by 0xAplki** [comment]
pls resolve conflicts

**Review by 0xAplki — CHANGES_REQUESTED** [review]
Pls resolve conflicts

File: markets/halo-matic/commons.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    [ePolygonNetwork.mumbai]: '0x943E44157dC0302a5CEb172374d1749018a00994',
+    [ePolygonNetwork.matic]: '0xD7D86236d6c463521920fCC50A9CB56f8C8Bf008',
+  },
+  LendingRateOracle: {
+    [ePolygonNetwork.mumbai]: '0xC661e1445F9a8E5FD3C3dbCa0A0A2e8CBc79725D',
+    [ePolygonNetwork.matic]: '0x17F73aEaD876CC4059089ff815EDA37052960dFB',
+  },
+  LendingPoolCollateralManager: {
+    [ePolygonNetwork.mumbai]: '0x2A7004B21c49253ca8DF923406Fed9a02AA86Ba0',
+    [ePolygonNetwork.matic]: '0xA39599424642D9fD35e475EF802EddF798dc555B',
```
Comment (tracyarciaga) [inline markets/halo-matic/commons.ts:97]:
```suggestion
    [ePolygonNetwork.matic]: '0xeEc23b994F27B6914A06180c6B5FA4f83ec80712',
```

File: markets/halo-matic/commons.ts [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+    [ePolygonNetwork.mumbai]: '0xE6ef11C967898F9525D550014FDEdCFAB63536B5',
+    [ePolygonNetwork.matic]: '0x3ac4e9aa29940770aeC38fe853a4bbabb2dA9C19',
+  },
+  ProviderRegistryOwner: {
+    [ePolygonNetwork.mumbai]: '0x943E44157dC0302a5CEb172374d1749018a00994',
+    [ePolygonNetwork.matic]: '0xD7D86236d6c463521920fCC50A9CB56f8C8Bf008',
+  },
+  LendingRateOracle: {
+    [ePolygonNetwork.mumbai]: '0xC661e1445F9a8E5FD3C3dbCa0A0A2e8CBc79725D',
+    [ePolygonNetwork.matic]: '0x17F73aEaD876CC4059089ff815EDA37052960dFB',
```
Comment (tracyarciaga) [inline markets/halo-matic/commons.ts:93]:
```suggestion
    [ePolygonNetwork.matic]: '0x5AE7Ef3118D3F8739BA5525534367757FEd03540',
```

File: markets/halo-matic/constants.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { ZERO_ADDRESS } from '../../helpers/constants';
+import { ePolygonNetwork } from '../../helpers/types';
+
+export const HALO_CONTRACT_ADDRESSES = {
+  [ePolygonNetwork.matic]: {
+    rewardToken: '0x47BE779De87de6580d0548cde80710a93c502405',
+    emissionManager: '0x235A2ac113014F9dcb8aBA6577F20290832dDEFd',
+    lendingPoolAddress: '',
```
Comment (tracyarciaga) [inline markets/halo-matic/constants.ts:8]:
```suggestion
    lendingPoolAddress: '0x51485c7278fa37f0f504F541f2755abda7715abf',
```

File: markets/halo-matic/commons.ts [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+  },
+  PoolAdminIndex: 0,
+  EmergencyAdminIndex: 0,
+  EmergencyAdmin: {
+    [ePolygonNetwork.mumbai]: undefined,
+    [ePolygonNetwork.matic]: undefined,
+  },
+  LendingPool: {
+    [ePolygonNetwork.mumbai]: '',
+    [ePolygonNetwork.matic]: '',
```
Comment (tracyarciaga) [inline markets/halo-matic/commons.ts:77]:
```suggestion
    [ePolygonNetwork.matic]: '0x51485c7278fa37f0f504F541f2755abda7715abf',
```

File: markets/halo-matic/commons.ts [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+  },
+  LendingPoolConfigurator: {
+    [ePolygonNetwork.mumbai]: '',
+    [ePolygonNetwork.matic]: '',
```
Comment (tracyarciaga) [inline markets/halo-matic/commons.ts:81]:
```suggestion
    [ePolygonNetwork.matic]: '0xBacdf66F391E9CB65C1D7CD3e506b6AEE5fA2F88',
```

File: markets/halo-matic/commons.ts [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+  },
+  ProviderRegistry: {
+    [ePolygonNetwork.mumbai]: '0xE6ef11C967898F9525D550014FDEdCFAB63536B5',
+    [ePolygonNetwork.matic]: '0x3ac4e9aa29940770aeC38fe853a4bbabb2dA9C19',
```
Comment (tracyarciaga) [inline markets/halo-matic/commons.ts:85]:
```suggestion
    [ePolygonNetwork.matic]: '0x2Ee7e1FE1B65600A3bD28B06d0b958B3b6cc4695',
```

**Review by 0xAplki — APPROVED** [review]

**Review by 0xAplki — APPROVED** [review]

**Pull request merged**